server.js

```diff
@@ -1,38 +1,140 @@
+3. Creating a new instance based on the Books mongoose model above and the books.json array of data which is being iterated over and saved to the mongoDB database.
+4. If statement will be true if I have entered the environment variable RESET_DB and equalled it to true when doing npm run dev in the terminal. This will signify that I want the database to be re-run and that the seeding of the database will begin!
+*/
+if(process.env.RESET_DB) {
+  const seedDatabase = async () => {
+  // deleteMany will allow for the data that was stored on the database on the last get request to be deleted so that there are no dupications of data. Then the new Authors will be created again.
+    await Book.deleteMany({});
+  
+    bookData.forEach((bookData) => {
+      new Book(bookData).save();
+    })
+  };
+  
+  seedDatabase();
+}
+
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(bodyParser.json());
+
+// First endpoint that will return the whole array of books data
+// Async and await are being used as find is asyncronous and might take some time to return the data being fetched from the server.
+app.get('/books', async (request, response) => {
+  const {author, title, language, averagerating } = request.query;
+
+  // RegExp is a regular expression object that is used to match the author string/text added to the query parameter with a pattern. 
+  if(author) {
+    const authorRegexp = new RegExp(author, "i");
+    const queryAuthor = await Book.find({authors: authorRegexp});
+    if(queryAuthor.length === 0){
+      return response.status(404).json({ error: "Sorry that we don't have any data for that author"});
+    }
+    return response.json(queryAuthor);    
+  }
+
+  if(title) {
+    const titleRegexp = new RegExp(title, "i");
+    const queryTitle = await Book.find({title: titleRegexp});
+    if(queryTitle.length === 0){
+      return response.status(404).json({ error: "Sorry that we don't have any data for that title"});
+    }
+    return response.json(queryTitle);    
+  }
+
+  if(language) {
+    const langRegexp = new RegExp(language, "i");
+    const queryLanguage = await Book.find({language_code: langRegexp});
+    if(queryLanguage.length === 0){
+      return response.status(404).json({ error: "Sorry that we don't have any data for that language"});
+    }
+    return response.json(queryLanguage);    
+  }
+
+  if(averagerating) {
+    const queryAveRat = await Book.find({average_rating: averagerating});
+    if(queryAveRat.length === 0){
+      return response.status(404).json({ error: "Sorry that we don't have any data for that average rating"});
+    }
+    return response.json(queryAveRat);    
+  }
 
-// Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
+  const books = await Book.find();
+  response.json(books);
 })
 
```

Comment:
I see you use nested if's instead of else. Why is that better? Code coach: is there a convention here when to use what?